### DISCUSSION TITLE
Stride Project - Change name (Namespace)

### DISCUSSION THREAD

**Discussion by Generalec** [stride3d/stride#2203] [Q&A] [answered]
Hello,

I started development of project under default name - MyGame4.

Now I advaced with project and it would be great to rename project to suitable name of it.

What is the best approach to change project name?

It's better start new project and copy all stuff?

Thank you.

**Comment by IXLLEGACYIXL** [comment]
its easier to copy it to a new project
else the project name changing isnt different to normal c#

**Reply by Doprez** [reply, marked as answer]
The main difference will come with any serialized assets from Components in the Scene.

I have used a find all and replace in VS Code to change namespaces before with success but its definitely easier to transfer to a new project assuming the rename doesnt work flawlessly.

In the future for any classes that are seen in the editor you should add a DataContract to the class with the class name as the passed in parameter IE `[DataContract(nameof(MyClass))]` This will avoid it using the namespace as a reference in the yaml files.
